Public Record Office Victoria.js

```diff
@@ -0,0 +1,216 @@
+	Copyright © 2024 Tim Sherratt
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+async function detectWeb(doc, url) {
+	if (getSearchResults(doc, true)) {
+		return "multiple";
+	// match items but not series, agencies, or functions (VPRS, VA or VF)
+	}
+	else if (url.match(/archive\/(?!VPRS|VA|VF)[A-Z0-9-]+/)) {
+		return "manuscript";
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	let items = {};
+	let found = false;
+	let rows = doc.querySelectorAll(".sub_heading_search_result");
+	for (let row of rows) {
+		let href = row.href;
+		href = href.match(/archive\/(?!VPRS|VA|VF)[A-Z0-9-]+/) ? href : null;
+		let title = row.innerText.replace(/\n/g, " ");
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (await detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (items) {
+			for (let url of Object.keys(items)) {
+				await scrape(url);
+			}
+		}
+	}
+	else {
+		await scrape(url);
+	}
+}
+
+async function scrape(url) {
+	// Get the API record for this item
+	let id = url.match(/archive\/([A-Z0-9-]+)/)[1];
+	let apiURL = "https://api.prov.vic.gov.au/search/query?wt=json&q=(_id%3A" + id + ")";
+	let apiJSON = await requestJSON(apiURL);
+	let record = apiJSON.response.docs[0];
+	
+	// Create the Zotero item
+	let item = new Zotero.Item("manuscript");
+	item.title = record.title;
+	item.type = record.category;
+	item.archive = "Public Record Office Victoria";
+	item.archiveLocation = record.citation;
+	item.libraryCatalog = "PROV API";
+	item.url = url;
+	item.rights = record.rights_status.join(", ");
+	item.place = record.location.join(", ");
+	item.abstractNote = record["description.aggregate"];
+
+	// Normalise dates and drop default values
+	let startDate = record.start_dt.replace("T00:00:00Z", "");
+	// Discard default
+	startDate = startDate != "1753-01-01" ? startDate : "";
+	let endDate = record.end_dt.replace("T00:00:00Z", "");
+	// Discard default
+	endDate = endDate != "3000-12-31" ? endDate : "";
+
+	// If there's a date range use 'issued', otherwise use 'date'
+	if (startDate == endDate) {
+		item.date = startDate;
+	}
+	else {
+		item.issued = startDate + "/" + endDate;
```

Comment:
@AbeJellinek -- could you weigh in on this. This will work (i.e. write the string to Extra), but I'm not sure how comfortable we are using this behavior in translators.

Reply:
I was going on what has been recommended in the forums for date ranges. Eg: https://forums.zotero.org/discussion/63073/date-year-range and https://forums.zotero.org/discussion/105949/date-range

Reply:
Right, I was referring to using item.issued to putting this into Extra -- the approach to date ranges using `Extra` is definitely right.

I kind of like the less verbose version using `item.issued`  -- is there actually a reason not to do this? will also be much faster to convert if we do get a better date field.

Reply:
I've changed all the `extra` additions to use the more verbose form. I also changed `series` to `Archive Collection` to be in line with CSL.